### ISSUE TITLE
Investigate life-cycle of c core native channel vs HTTP/2 HttpClient

### ISSUE THREAD

**Issue by JamesNK** [grpc/grpc-dotnet#157]
Figure out life-cycle and performance differences of c core Channel vs HttpClient.

i.e. open sockets, starting/stopping

Also figure out whether HttpClientFactory should pool HttpClient.

**Comment by JamesNK** [comment]
Related https://github.com/grpc/grpc-dotnet/issues/134

**Comment by JamesNK** [comment]
Channels specs - https://github.com/grpc/grpc-dotnet/pull/180#discussion_r269667135

**Comment by JamesNK** [comment]
Some thoughts on Channel in Grpc.Core ([Grpc.Core.Channel](https://grpc.github.io/grpc/csharp/api/Grpc.Core.Channel.html)) and other languages, and what to do in grpc-dotnet:

1. Channel in Grpc.Core is mostly a thin wrapper around the underlying C-Core implementation. For example, most settings are set via untyped name/value arguments: `["grpc.max_message_length"] = 1024`. It's functional, but not ideal.
2. Java has a strongly typed channel API - https://grpc.github.io/grpc-java/javadoc/io/grpc/ManagedChannelBuilder.html. I think designing a new, strongly typed Channel configuration like this would give users a better experience.
3. I don't see much value in keeping compatibility with most (all?) of the existing Channel types. It will not be possible to allow users to take an existing gRPC client apps to our new packages with zero code changes. I think the client should follow what the server has done: replacing startup and configuration, and keep compatibility of generation code from proto files.
4. Is there any value in introducing ChannelBase? The code gen for LiteClientBase could be extended to include a generated constructor that takes a ChannelBase. If we did that then ChannelBase would need to provide a method to create a call invoker from itself.
I moved methods that I think would be universally useful from Channel. These line up with what is available in Java's [ManagedChannel](https://grpc.github.io/grpc-java/javadoc/io/grpc/ManagedChannel.html).
```cs
public abstract class ChannelBase
{
    // Moved from Channel
    public Task ConnectAsync(DateTime? deadline = null);
    public Task ShutdownAsync();
    public Task<bool> TryWaitForStateChangedAsync(ChannelState lastObservedState, DateTime? deadline = null);
    public Task WaitForStateChangedAsync(ChannelState lastObservedState, DateTime? deadline = null);
    public CancellationToken ShutdownToken { get; }
    public ChannelState State { get; }
    public string Target { get; }

    // Used by generated client constructor?
    public CallInvoker CreateCallInvoker();

    // Protected abstract methods that are called by the public methods
    protected abstract Task ConnectAsyncCore(DateTime? deadline = null);
    // ShutdownAsyncCore, TryWaitForStateChangedAsyncCore, etc
}
```
```cs
public LiteClientBase(ChannelBase channel) : this(channel.CreateCallInvoker())
{
}
```
5. I'm going to talk with .NET networking about whether Connect/Shutdown like functionality could be added in the Http stack. If ConnectAsync/ShutdownAsync was added to a grpc-dotnet channel API they wouldn't do anything in 3.0. Maybe ever?
6. Because ConnectAsync et al won't do anything in 3.0, we could leave them off ChannelBase for now. ChannelBase could simply be:
```cs
public abstract class ChannelBase
{
    public string Target { get; }
    protected abstract string TargetCode { get; }

    public CallInvoker CreateCallInvoker();
    protected abstract CallInvoker CreateCallInvokerCore();
}
```
7. I don't think any breaking changes are required in Channel for what I've discussed.

@jtattermusch @JunTaoLuo @shirhatti @davidfowl
